### PR TITLE
refactor(config): load specific service config

### PR DESCRIPTION
**Pull request by arijitdey**
## Description
Load specific service config instead of uber config

## How Has This Been Tested?
Added a unit test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd.go [hunk 1]
```diff
@@ -152,32 +160,29 @@ func (r *Runner) runE(runner interface{}, cfg interface{}) func(cmd *cobra.Comma
...
+					logging.WithLogger(l),
+					FxContext(ctx),
+					fx.Supply(cmd),
+					fx.Supply(service.Name(r.prefix)),
+					fx.Supply(logging.Level(loglevelFlag)),
+					Module,
+					module,
+				)
+
+				//populate() error
```
Comment (sethyates) [inline cmd.go:174]: Delete

File: cmd.go [hunk 2]
```diff
@@ -137,13 +138,20 @@ func (r *Runner) preRunE(cmd *cobra.Command, args []string) error {
 	return r.prevPreRunE(cmd, args)
 }
 
-func (r *Runner) runE(runner interface{}, cfg interface{}) func(cmd *cobra.Command, args []string) error {
+func (r *Runner) runE(runner interface{}) func(cmd *cobra.Command, args []string) error {
 	return func(cmd *cobra.Command, args []string) error {
-		// Unmarshal the configuration
-		if err := Unmarshal(r.prefix, cfg); err != nil {
-			return errors.Wrap(err, "unable to unmarshal configuration")
+		envFlag, err := cmd.Flags().GetString("env")
```
Comment (sethyates) [inline cmd.go:143]: This is a bad merge. This was removed recently. Rebase.

File: cmd.go [hunk 3]
```diff
@@ -137,13 +138,20 @@ func (r *Runner) preRunE(cmd *cobra.Command, args []string) error {
...
+		if err != nil {
+			return err
 		}
 
+		ctx, span := tracer.Start(cmd.Context(), "Run")
```
Comment (sethyates) [inline cmd.go:148]: Bad merge

File: fx.go [hunk 4]
```diff
@@ -22,23 +22,18 @@ package orlop
...
-func FxOptions(o config.Config) fx.Option {
-	return o.Options()
-}
-
 func FxContext(ctx context.Context) fx.Option {
 	return fx.Provide(func() context.Context { return ctx })
 }
 
-func Populate(ctx context.Context, prefix string, e env.Environment, module fx.Option, cfg config.Config, targets ...interface{}) error {
+func Populate(ctx context.Context, prefix string, e env.Environment, module fx.Option, cfg interface{}, targets ...interface{}) error {
```
Comment (sethyates) [inline fx.go:36]: Remove cfg. May need a way of injecting an environ instead.

**Review by sethyates — CHANGES_REQUESTED** [review]

File: cmd_test.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+				)
+			},
+		),
+	)
+
+	//TODO: This is not ideal
+	go Run("test", module)
+
+	time.Sleep(1 * time.Second)
+
```
Comment (arijitdey) [inline cmd_test.go:62]: Not sure about how to write a better test using `fxtest`. Any suggestions? 

In the meantime, will keep tinkering around.

File: cmd.go [hunk 6]
```diff
@@ -137,59 +139,33 @@ func (r *Runner) preRunE(cmd *cobra.Command, args []string) error {
...
+			fx.Supply(cmd),
+			fx.Supply(service.Name(r.prefix)),
+			fx.Supply(logging.Level(loglevelFlag)),
+			Module,
+			module,
+			fx.Invoke(
+				func(provider config.Provider) {
+					provider.Load(cmd.Context())
+				},
+			),
```
Comment (arijitdey) [inline cmd.go:163]: Curious - If this is the correct way to call `Load`?

Comment (sethyates) [reply cmd.go:163]: No, inject the ctx as well, so it would be:
```
func(ctx context.Context, provider config.Provider) error {
	return provider.Load(ctx)
},
```
And actually this function could probably be moved to the config/module.go and you should return an error at least

File: cmd.go [hunk 7]
```diff
@@ -237,14 +213,14 @@ func (r *Runner) SetupLogging(env Environment, loglevel string) {
 }
 
 // Run loads config and then executes the given runner
-func Run(prefix string, runner interface{}, cfg interface{}) {
+func Run(prefix string, runner fx.Option) {
```
Comment (sethyates) [inline cmd.go:216]: To keep this from being a breaking change, let's add back `, _ interface{}`

File: cmd_test.go [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+						OnStart: func(context.Context) error {
+							return nil
+						},
+						OnStop: func(context.Context) error {
+							return nil
+						},
+					},
+				)
+			},
+		),
```
Comment (sethyates) [inline cmd_test.go:55]: Why?

File: cmd_test.go [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+	)
+
+	//TODO: This is not ideal
+	go Run("test", module)
```
Comment (sethyates) [inline cmd_test.go:59]: Why are you doing a go routine?

File: config/provider.go [hunk 10]
```diff
@@ -20,6 +20,15 @@
...
+	"context"
+
+	"go.ketch.com/lib/orlop/v2/service"
+)
+
 type Provider interface {
-	Load(cfg Config) error
+	Register(ctx context.Context, service string, cfg interface{})
+	Get(ctx context.Context, service string) (interface{}, error)
+	List(ctx context.Context, prefix service.Name) ([]string, error)
```
Comment (sethyates) [inline config/provider.go:32]: The prefix should be injected into the constructor of the implementation.

File: cmd_test.go [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+				return c.(*TestConfig), nil
+			},
+		),
+		fx.Invoke(
+			func(provider config.Provider) {
+				provider.Register(context.Background(), "config", &cfg)
+			},
+		),
+		fx.Invoke(
+			func(lifecycle fx.Lifecycle, config *TestConfig) {
```
Comment (sethyates) [inline cmd_test.go:43]: You could take a `fx.Shutdown` and call `shutdown` on the `OnStart`.

File: cmd.go [hunk 12]
```diff
@@ -77,17 +75,21 @@ func (r *Runner) SetupRoot(cmd *cobra.Command) *Runner {
...
 	}
 
 	cmd.AddCommand(&cobra.Command{
 		Use:   "init",
 		Short: "output the config environment variables and exits",
 		Args:  cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			vars, err := GetVariablesFromConfig(r.prefix, cfg)
+			var cfgMgr config.Provider
+			app := fx.New(Module, fx.Populate(&cfgMgr))
```
Comment (sethyates) [inline cmd.go:89]: Inject the `service.Name` here.

File: cmd.go [hunk 13]
```diff
@@ -77,17 +75,21 @@ func (r *Runner) SetupRoot(cmd *cobra.Command) *Runner {
 }
 
 // Setup sets up the Command
-func (r *Runner) Setup(cmd *cobra.Command, runner interface{}, cfg interface{}) *Runner {
+func (r *Runner) Setup(cmd *cobra.Command, runner fx.Option) *Runner {
```
Comment (sethyates) [inline cmd.go:78]: To keep this from being a breaking change, let's add back `, _ interface{}`

Comment (arijitdey) [reply cmd.go:78]: Added the `_ interface{}` back in `Run(..)`. As `Setup` is called by `Run`, we might not need it here. Or is it being used in other places? 🤔

File: cmd.go (hunk 12 above)
Comment (arijitdey) [reply cmd.go:89]: Using the prefix from Environ.

File: cmd.go [hunk 14]
```diff
@@ -137,59 +139,28 @@ func (r *Runner) preRunE(cmd *cobra.Command, args []string) error {
...
-			}
-		}
+		app := fx.New(
+			logging.WithLogger(l),
+			FxContext(cmd.Context()),
+			fx.Supply(cmd),
+			fx.Supply(service.Name(r.prefix)),
+			fx.Supply(logging.Level(loglevelFlag)),
+			module,
+			Module,
```
Comment (arijitdey) [inline cmd.go:158]: Is it safe to re-order this? This to make sure that the `Load(..)` is invoked prior to all other invokes.

Comment (sethyates) [reply cmd.go]: No it is not safe because invoker hooks are registered in order.

File: cmd.go (hunk 13 above)
Comment (sethyates) [reply cmd.go:78]: Yes it is used in other places.

File: cmd.go (hunk 12 above)
Comment (sethyates) [reply cmd.go:89]: How is that getting injected then?

File: cmd_test.go [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+				return c.(*TestConfig), nil
+			},
+		),
+		fx.Invoke(
+			func(ctx context.Context, provider config.Provider) {
+				provider.Register(ctx, "config", &cfg)
+			},
+		),
+		fx.Invoke(
+			func(lifecycle fx.Lifecycle, s fx.Shutdowner, config *TestConfig) {
```
Comment (sethyates) [inline cmd_test.go:43]: Don't need config.

File: cmd_test.go [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+	var cfg TestConfig
+
+	var module = fx.Options(
+		fx.Provide(
+			func(ctx context.Context, provider config.Provider) (*TestConfig, error) {
+				c, err := provider.Get(ctx, "config")
+				if err != nil {
+					return nil, err
+				}
+				return c.(*TestConfig), nil
```
Comment (sethyates) [inline cmd_test.go:34]: Should we standardise on using non-pointer config objects?

File: config/impl.go [hunk 17]
```diff
@@ -21,54 +21,85 @@
...
+		configs: make(map[string]interface{}, 0),
 		environ: environ,
 	}
 }
 
-func (s *providerImpl) Load(cfg Config) error {
-	fields, err := reflectStruct([]string{}, cfg)
-	if err != nil {
-		return err
+func (s *providerImpl) Register(_ context.Context, service string, cfg interface{}) {
```
Comment (sethyates) [inline config/impl.go:48]: I am now thinking perhaps we should switch to a more declarative interface, such that we can do the following:
```
fx.Supply(config.Config("server", &Config)),
```

Then rely on an optional group of these `config.ConfigDef` structs when creating this provider.

File: config/impl.go [hunk 18]
```diff
@@ -21,54 +21,85 @@
...
-		} else if field.tag.DefaultValue != nil {
-			if err = field.set(field.v, *field.tag.DefaultValue); err != nil {
-				return errors.Wrapf(err, "failed to set field '%s' with value '%s'", name, v)
+	return nil, errors.Errorf("%s config not found", service)
+}
+
+func (s *providerImpl) List(_ context.Context) ([]string, error) {
+	var vars []string
+	for k, v := range s.configs {
+		key := strcase.ToScreamingSnake(strings.Join([]string{string(s.environ.GetPrefix()), k}, "_"))
```
Comment (sethyates) [inline config/impl.go:63]: This seems to be used in a few places - perhaps we make a utility function?

File: config/impl.go [hunk 19]
```diff
@@ -21,54 +21,85 @@
...
+
+func (s *providerImpl) Load(_ context.Context) error {
+	for k, v := range s.configs {
+		fields, err := reflectStruct([]string{}, v)
+		if err != nil {
+			return err
+		}
+
+		for name, field := range fields {
+			keyName := strings.Join([]string{k, name}, "_")
```
Comment (sethyates) [inline config/impl.go:83]: This is done differently to above.

File: config/module.go [hunk 20]
```diff
@@ -21,11 +21,26 @@
 package config
 
 import (
+	"context"
+
 	"go.uber.org/fx"
 )
 
+func Lifecycle(lc fx.Lifecycle, provider Provider) {
```
Comment (sethyates) [inline config/module.go:29]: Move to a `lifecycle.go`

File: config/provider.go [hunk 21]
```diff
@@ -20,6 +20,13 @@
...
 package config
 
+import (
+	"context"
+)
+
 type Provider interface {
-	Load(cfg Config) error
+	Register(ctx context.Context, service string, cfg interface{})
+	Get(ctx context.Context, service string) (interface{}, error)
```
Comment (sethyates) [inline config/provider.go:29]: Given we now have go 1.18, I wonder if this interface{} could be replaced with a `<T>`

File: env/environ.go [hunk 22]
```diff
@@ -44,3 +45,7 @@ type environImpl struct {
 func (e environImpl) Getenv(key string) string {
 	return os.Getenv(strcase.ToScreamingSnake(strings.Join([]string{string(e.prefix), key}, "_")))
 }
+
+func (e environImpl) GetPrefix() service.Name {
```
Comment (sethyates) [inline env/environ.go:49]: This shouldn't be exposed.

File: env/environ.go [hunk 23]
```diff
@@ -29,6 +29,7 @@ import (
 
 type Environ interface {
 	Getenv(key string) string
+	GetPrefix() service.Name
```
Comment (sethyates) [inline env/environ.go:32]: This is an unnecessary wart on the interface.

File: config/impl.go (hunk 18 above)
Comment (sethyates) [inline config/impl.go:63]: You should inject the prefix rather than relying on the environ.

File: config/provider.go (hunk 21 above)
Comment (arijitdey) [reply config/provider.go:29]: Yup. Replaced with `any`.

File: cmd_test.go (hunk 16 above)
Comment (arijitdey) [reply cmd_test.go:34]: May be not as we can't return a nil value in case of an error.

Comment (sethyates) [reply cmd_test.go:40]: In this case you could return `TestConfig{}, err`

File: cmd.go [hunk 24]
```diff
@@ -77,17 +75,21 @@ func (r *Runner) SetupRoot(cmd *cobra.Command) *Runner {
...
 	}
 
 	cmd.AddCommand(&cobra.Command{
 		Use:   "init",
 		Short: "output the config environment variables and exits",
 		Args:  cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			vars, err := GetVariablesFromConfig(r.prefix, cfg)
+			var cfgMgr config.Provider
+			app := fx.New(Module, fx.Populate(&cfgMgr))
```
Comment (sethyates) [inline cmd.go:89]: We'll have to include the runner module here too since it is the one that adds all the ConfigDefinitions.

File: cmd.go [hunk 25]
```diff
@@ -77,17 +75,21 @@ func (r *Runner) SetupRoot(cmd *cobra.Command) *Runner {
...
+			app.Run()
```
Comment (sethyates) [inline cmd.go:90]: Does this work? No shutdown has been injected above.

File: cmd.go [hunk 26]
```diff
@@ -137,59 +139,28 @@ func (r *Runner) preRunE(cmd *cobra.Command, args []string) error {
...
 }
 
-func (r *Runner) runE(runner interface{}, cfg interface{}) func(cmd *cobra.Command, args []string) error {
+func (r *Runner) runE(module fx.Option) func(cmd *cobra.Command, args []string) error {
 	return func(cmd *cobra.Command, args []string) error {
-		// Unmarshal the configuration
-		if err := Unmarshal(r.prefix, cfg); err != nil {
-			return errors.Wrap(err, "unable to unmarshal configuration")
-		}
-
```
Comment (sethyates) [inline cmd.go:146]: For backwards compatibility, I think we'll need to add this back for now (maybe mark to be removed at some point in the future.

File: cmd.go [hunk 27]
```diff
@@ -137,59 +139,28 @@ func (r *Runner) preRunE(cmd *cobra.Command, args []string) error {
...
 			return err
 		}
 
-		if module, ok := runner.(fx.Option); ok {
-			if _, ok = cfg.(config.Config); ok {
-				runner = func(ctx context.Context, cfg config.Config) error {
-					app := fx.New(
-						logging.WithLogger(l),
-						FxContext(ctx),
-						FxOptions(cfg),
```
Comment (sethyates) [inline cmd.go:160]: We'll also need to add this back to keep from breaking everything existing.

File: config/impl.go [hunk 28]
```diff
@@ -21,54 +21,88 @@
...
-}
-
 type providerImpl struct {
+	configs map[string]any
 	environ env.Environ
+	prefix  service.Name
 }
 
-func New(environ env.Environ) Provider {
+func New(environ env.Environ, prefix service.Name, p Params) Provider {
```
Comment (sethyates) [inline config/impl.go:42]: If you have `Params`, why not move all of the required Params there instead of leaving `environ` and `prefix` here?

File: config/impl.go [hunk 29]
```diff
@@ -21,54 +21,88 @@
...
-		} else if field.tag.DefaultValue != nil {
-			if err = field.set(field.v, *field.tag.DefaultValue); err != nil {
-				return errors.Wrapf(err, "failed to set field '%s' with value '%s'", name, v)
+	return nil, errors.Errorf("%s config not found", service)
+}
+
+func (s *providerImpl) List(_ context.Context) ([]string, error) {
+	var vars []string
+	for k, v := range s.configs {
+		key := strcase.ToScreamingSnake(strings.Join([]string{string(s.prefix), k}, "_"))
```
Comment (sethyates) [inline config/impl.go:66]: move this to a helper.

File: config/params.go [hunk 30]
```diff
@@ -0,0 +1,9 @@
+package config
+
+import "go.uber.org/fx"
+
+type Params struct {
+	fx.In
+
+	Defs []Definition `optional:"true"`
```
Comment (sethyates) [inline config/params.go:8]: `group`

File: fx.go [hunk 31]
```diff
@@ -22,33 +22,23 @@ package orlop
...
-func FxOptions(o config.Config) fx.Option {
-	return o.Options()
-}
-
 func FxContext(ctx context.Context) fx.Option {
 	return fx.Provide(func() context.Context { return ctx })
 }
 
-func Populate(ctx context.Context, prefix string, e env.Environment, module fx.Option, cfg config.Config, targets ...interface{}) error {
+func Populate(ctx context.Context, prefix string, e env.Environment, module fx.Option, targets ...interface{}) error {
```
Comment (sethyates) [inline fx.go:36]: Let's revert these changes to reduce the blast radius of this change.

File: cmd.go (hunk 25 above)
Comment (arijitdey) [reply cmd.go:90]: You are right added the shutdown to lifecycle onStart

File: config/impl.go [hunk 32]
```diff
@@ -21,57 +21,94 @@
...
-			return errors.Errorf("%s required", name)
 		}
 	}
 
 	return nil
 }
 
 // GetVariablesFromConfig returns the environment variables from the given config object
-func GetVariablesFromConfig(prefix service.Name, cfg Config) ([]string, error) {
+func GetVariablesFromConfig(prefix service.Name, service string, cfg any) ([]string, error) {
```
Comment (arijitdey) [inline config/impl.go:108]: Curious - If changing the arguments would break this in other services that import orlop.

File: cmd.go [hunk 33]
```diff
@@ -87,7 +88,26 @@ func (r *Runner) Setup(cmd *cobra.Command, runner interface{}, cfg interface{})
 		Short: "output the config environment variables and exits",
 		Args:  cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			vars, err := GetVariablesFromConfig(r.prefix, cfg)
+			var cfgMgr config.Provider
+			app := fx.New(
+				Module,
+				runner,
+				fx.Populate(&cfgMgr),
+				fx.Invoke(
```
Comment (sethyates) [inline cmd.go:96]: I would move this above `runner`. We should test to ensure the `runner`'s lifecycles don't get executed (such as `server.Lifecycle`).

File: cmd.go [hunk 34]
```diff
@@ -77,7 +78,7 @@ func (r *Runner) SetupRoot(cmd *cobra.Command) *Runner {
 }
 
 // Setup sets up the Command
-func (r *Runner) Setup(cmd *cobra.Command, runner interface{}, cfg interface{}) *Runner {
+func (r *Runner) Setup(cmd *cobra.Command, runner fx.Option, cfg any) *Runner {
```
Comment (sethyates) [inline cmd.go:81]: Rename `runner` to `module` for consistency.

File: config/impl.go (hunk 32 above)
Comment (sethyates) [reply config/impl.go:108]: I didn't find any references to it outside of orlop. Perhaps we could even make it hidden.

File: config/params.go [hunk 35]
```diff
@@ -0,0 +1,27 @@
...
+type Definition struct {
+	Name   string
+	Config any
+}
+
+func ConfigOption(name string, config any) fx.Option {
+	return fx.Supply(fx.Annotate(Definition{
+		Name:   name,
+		Config: config,
+	}, fx.ResultTags(`group:"configs"`)))
```
Comment (sethyates) [inline config/params.go:18]: We tend to format more openly like this:
```
	return fx.Supply(
		fx.Annotate(
			Definition{
				Name:   name,
				Config: config,
			},
			fx.ResultTags(`group:"configs"`),
		),
	)
```

File: cmd.go (hunk 33 above)
Comment (arijitdey) [reply cmd.go:96]: Will look into the testing bits for init.

File: config/impl.go [hunk 36]
```diff
@@ -21,57 +21,94 @@
...
 		}
 	}
 
 	return nil
 }
 
-// GetVariablesFromConfig returns the environment variables from the given config object
-func GetVariablesFromConfig(prefix service.Name, cfg Config) ([]string, error) {
+// getVariablesFromConfig returns the environment variables from the given config object
+func getVariablesFromConfig(prefix service.Name, service string, cfg any) ([]string, error) {
```
Comment (sethyates) [inline config/impl.go:108]: If this is not used anywhere else, you can make a member function of the struct and would then have prefix available there.

File: cmd/cmd_test.go [hunk 37]
```diff
@@ -0,0 +1,114 @@
...
+		Use:              "test",
+		TraverseChildren: true,
+		SilenceUsage:     true,
+	}
+
+	NewRunner("test").SetupRoot(cmd).Setup(cmd, module)
+
+	cmd.SetArgs([]string{"init"})
+	err := cmd.Execute()
+	require.NoError(t, err)
```
Comment (arijitdey) [inline cmd/cmd_test.go:113]:
```
=== RUN   TestInit
[Fx] PROVIDE	context.Context <= go.ketch.com/lib/orlop/v2.FxContext.func1()
[Fx] SUPPLY	service.Name
%!(EXTRA string=)[Fx] PROVIDE	config.Provider <= go.ketch.com/lib/orlop/v2/config.New()
[Fx] PROVIDE	env.Environment <= go.ketch.com/lib/orlop/v2/env.Env()
[Fx] PROVIDE	env.Environ <= go.ketch.com/lib/orlop/v2/env.NewEnviron()
[Fx] PROVIDE	logging.Logger <= go.ketch.com/lib/orlop/v2/logging.New()
[Fx] PROVIDE	parameter.ObjectStore <= go.ketch.com/lib/orlop/v2/parameter.NewObjectStore()
[Fx] SUPPLY	config.Definition
%!(EXTRA string=)[Fx] PROVIDE	cmd.TestConfig <= go.ketch.com/lib/orlop/v2/cmd.TestInit.func1()
[Fx] PROVIDE	fx.Lifecycle <= go.uber.org/fx.New.func1()
[Fx] PROVIDE	fx.Shutdowner <= go.uber.org/fx.(*App).shutdowner-fm()
[Fx] PROVIDE	fx.DotGraph <= go.uber.org/fx.(*App).dotGraph-fm()
[Fx] INVOKE		reflect.makeFuncStub()
[Fx] INVOKE		go.ketch.com/lib/orlop/v2/cmd.(*Runner).Setup.func1.1()
[Fx] HOOK OnStart		go.ketch.com/lib/orlop/v2/cmd.(*Runner).Setup.func1.1.1() executing (caller: go.ketch.com/lib/orlop/v2/cmd.(*Runner).Setup.func1.1)
[Fx] HOOK OnStart		go.ketch.com/lib/orlop/v2/cmd.(*Runner).Setup.func1.1.1() called by go.ketch.com/lib/orlop/v2/cmd.(*Runner).Setup.func1.1 ran successfully in 417ns
[Fx] RUNNING
[Fx] TERMINATED
#TEST_CONFIG_BASE64_ENCODED=# [v1, v2, v3]
TEST_CONFIG_CUSTOM=12345s
TEST_CONFIG_DEF=/pki/issue
TEST_CONFIG_EMBEDDED=false # bool
#TEST_CONFIG_HEX_ENCODED=# [v1, v2, v3]
#TEST_CONFIG_MAP=# [k=v, k=v, k=v]
TEST_CONFIG_PTR=
#TEST_CONFIG_REQ=# string
#TEST_CONFIG_SLICED=# [v1, v2, v3]
TEST_CONFIG_UNKNOWN=0 # int
--- PASS: TestInit (0.00s)
PASS
```

**Comment by sethyates** [comment]
See https://github.com/ketch-com/astrolabe/pull/66 for example of how this affects a service.

**Review by sethyates — APPROVED** [review]
LGTM
